File: TealiumBrazeExample/app/src/main/java/com/tealium/example/UserActivity.java
package com.tealium.example;

import android.app.DatePickerDialog;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.DatePicker;
import android.widget.EditText;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import com.tealium.example.helper.TealiumHelper;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

public class UserActivity extends AppCompatActivity {

    EditText mFirstNameEditText;
    EditText mLastNameEditText;
    EditText mEmailEditText;
    EditText mGenderEditText;
    EditText mHomeCityEditText;
    EditText mBirthdayEditText;
    EditText mUserIdEditText;
    EditText mUserAliasEditText;
    EditText mUserAliasLabelEditText;

    AppCompatButton mSaveButton;

    Calendar mCal;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user);

        mFirstNameEditText = findViewById(R.id.edit_txt_first_name);
        mLastNameEditText = findViewById(R.id.edit_txt_last_name);
        mEmailEditText = findViewById(R.id.edit_txt_email);
        mGenderEditText = findViewById(R.id.edit_txt_gender);
        mHomeCityEditText = findViewById(R.id.edit_txt_home_city);
        mBirthdayEditText = findViewById(R.id.edit_txt_birthday);
        mUserIdEditText = findViewById(R.id.edit_txt_user_id);
        mUserAliasEditText = findViewById(R.id.edit_txt_user_alias);
        mUserAliasLabelEditText = findViewById(R.id.edit_txt_user_alias_label);
        mSaveButton = findViewById(R.id.btn_save_user_details);

        mCal = Calendar.getInstance();
        final DatePickerDialog.OnDateSetListener date = new DatePickerDialog.OnDateSetListener() {

            @Override
            public void onDateSet(DatePicker view, int year, int monthOfYear,
                                  int dayOfMonth) {
                mCal.set(Calendar.YEAR, year);
                mCal.set(Calendar.MONTH, monthOfYear);
                mCal.set(Calendar.DAY_OF_MONTH, dayOfMonth);
                updateDateLabel();
            }

        };

        mBirthdayEditText.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new DatePickerDialog(UserActivity.this, date, mCal
                        .get(Calendar.YEAR), mCal.get(Calendar.MONTH),
                        mCal.get(Calendar.DAY_OF_MONTH)).show();
            }
        });

        mSaveButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Map<String, String> data = getFieldData();
                TealiumHelper.trackEvent("user_attribute", data);
                TealiumHelper.trackEvent("user_login", data);
                TealiumHelper.trackEvent("user_alias", data);
            }
        });

    }

    private void updateDateLabel() {
        String dataFormat = "dd/MM/yyyy";
        SimpleDateFormat sdf = new SimpleDateFormat(dataFormat, Locale.getDefault());

        mBirthdayEditText.setText(sdf.format(mCal.getTime()));
    }

    private Map<String, String> getFieldData() {
        Map<String, String> data = new HashMap<>();

        data.put("first_name", mFirstNameEditText.getText().toString());
        data.put("last_name", mLastNameEditText.getText().toString());
        data.put("email", mEmailEditText.getText().toString());
        String gender = mGenderEditText.getText().toString();
        if (TextUtils.isEmpty(gender)) {
            gender = "prefernotosay";
        }
        data.put("gender", gender);
        data.put("home_city", mHomeCityEditText.getText().toString());
        data.put("birthday", mBirthdayEditText.getText().toString());

        data.put("user_id", mUserIdEditText.getText().toString());
        data.put("user_alias", mUserAliasEditText.getText().toString());
        data.put("user_alias_label", mUserAliasLabelEditText.getText().toString());

        return data;
    }
}

File: app/src/main/java/com/tealium/example/UserActivity.kt
package com.tealium.example

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.os.Bundle
import android.text.TextUtils
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatButton
import com.tealium.example.helper.TealiumHelper.trackEvent
import java.text.SimpleDateFormat
import java.util.*

class UserActivity : AppCompatActivity() {

    private lateinit var firstNameEditText: EditText
    private lateinit var lastNameEditText: EditText
    private lateinit var emailEditText: EditText
    private lateinit var genderEditText: EditText
    private lateinit var homeCityEditText: EditText
    private lateinit var birthdayEditText: EditText
    private lateinit var userIdEditText: EditText
    private lateinit var userAliasEditText: EditText
    private lateinit var userAliasLabelEditText: EditText
    private lateinit var saveButton: AppCompatButton
    private lateinit var calendar: Calendar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user)
        firstNameEditText = findViewById(R.id.edit_txt_first_name)
        lastNameEditText = findViewById(R.id.edit_txt_last_name)
        emailEditText = findViewById(R.id.edit_txt_email)
        genderEditText = findViewById(R.id.edit_txt_gender)
        homeCityEditText = findViewById(R.id.edit_txt_home_city)
        birthdayEditText = findViewById(R.id.edit_txt_birthday)
        userIdEditText = findViewById(R.id.edit_txt_user_id)
        userAliasEditText = findViewById(R.id.edit_txt_user_alias)
        userAliasLabelEditText = findViewById(R.id.edit_txt_user_alias_label)
        saveButton = findViewById(R.id.btn_save_user_details)
        calendar = Calendar.getInstance()
        val date = OnDateSetListener { view, year, monthOfYear, dayOfMonth ->
            calendar.set(Calendar.YEAR, year)
            calendar.set(Calendar.MONTH, monthOfYear)
            calendar.set(Calendar.DAY_OF_MONTH, dayOfMonth)
            updateDateLabel()
        }
        birthdayEditText.setOnClickListener {
            DatePickerDialog(
                this@UserActivity, date, calendar
                    .get(Calendar.YEAR), calendar.get(Calendar.MONTH),
                calendar.get(Calendar.DAY_OF_MONTH)
            ).show()
        }
        saveButton.setOnClickListener {
            val data = fieldData
            trackEvent("user_attribute", data)
            trackEvent("user_login", data)
            trackEvent("user_alias", data)
        }
    }

    private fun updateDateLabel() {
        val dataFormat = "dd/MM/yyyy"
        val sdf = SimpleDateFormat(dataFormat, Locale.getDefault())
        birthdayEditText.setText(sdf.format(calendar.time))
    }

    private val fieldData: Map<String, String>
        get() = mapOf(
            "customer_first_name" to firstNameEditText.text.toString(),
            "customer_last_name" to lastNameEditText.text.toString(),
            "customer_email" to emailEditText.text.toString(),
            "customer_gender" to (if (TextUtils.isEmpty(genderEditText.text))
                "prefernotosay"
            else genderEditText.text.toString()),
            "customer_home_city" to homeCityEditText.text.toString(),
            "customer_dob" to birthdayEditText.text.toString(),
            "customer_id" to userIdEditText.text.toString(),
            "customer_alias" to userAliasEditText.text.toString(),
            "customer_alias_label" to userAliasLabelEditText.text.toString(),
        )

}